src/hotspot/share/oops/methodData.hpp

```diff
@@ -2310,7 +2311,9 @@ class MethodData : public Metadata {
   intx arg_local()                               { return _arg_local; }
   intx arg_stack()                               { return _arg_stack; }
   intx arg_returned()                            { return _arg_returned; }
-  uint arg_modified(int a)                       { ArgInfoData *aid = arg_info();
+  uint arg_modified(int a)                       { // Lock and avoid breaking lock with Safepoint
+                                                   MutexLocker ml(extra_data_lock(), Mutex::_no_safepoint_check_flag);
+                                                   ArgInfoData *aid = arg_info();
```

Comment:
Can you move these to methodData.inline.hpp so that you don't have to include mutexLocker.hpp in a header file?